modules/checkbox/react/spec/Checkbox.spec.tsx

```diff
@@ -1,102 +1,104 @@
+      expect(getByDisplayValue(value)).toBeDefined();
+    });
   });
 
-  test('should call a callback function', () => {
-    const component = mount(<Checkbox onChange={cb} />);
-    const input = component.find('input');
-    input.simulate('change');
-
-    expect(cb.mock.calls.length).toBe(1);
-    component.unmount();
+  describe('when rendered with checked=true', () => {
+    it('should render a checked checkbox input', () => {
+      const {getByRole} = render(<Checkbox checked={true} onChange={cb} />);
+      expect(getByRole('checkbox')).toHaveProperty('checked', true);
+    });
   });
 
-  test('Checkbox should spread extra props', () => {
-    const component = mount(<Checkbox data-propspread="test" onChange={cb} />);
-    const input = component
-      .find('input') // TODO: Standardize on prop spread location (see #150)
-      .getDOMNode();
-    expect(input.getAttribute('data-propspread')).toBe('test');
-    component.unmount();
+  describe('when rendered with disabled attribute', () => {
+    it('should render a disabled checkbox input', () => {
+      const {getByRole} = render(<Checkbox disabled={true} onChange={cb} />);
+      expect(getByRole('checkbox')).toHaveProperty('disabled', true);
+    });
   });
-});
 
-describe('Checkbox Accessibility', () => {
-  test('Checkbox should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <Checkbox id={'123'} label={'Label'} onChange={jest.fn()} />
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered without an id', () => {
+    it('should create a unique id for each instance', async () => {
+      const {getByLabelText} = render(
+        <form>
+          <Checkbox checked={true} onChange={cb} disabled={false} label="label1" />;
+          <Checkbox onChange={cb} disabled={false} label="label2" />;
+        </form>
+      );
+
+      const id1 = getByLabelText('label1').getAttribute('id');
+      const id2 = getByLabelText('label2').getAttribute('id');
+
+      expect(id1).not.toEqual(id2);
+    });
+
+    it('should keep the same unique id if re-rendered', () => {
+      const {getByRole, rerender} = render(<Checkbox checked={false} onChange={cb} />);
+
+      const uniqueId = getByRole('checkbox').getAttribute('id');
+      expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
+
+      rerender(<Checkbox checked={true} onChange={cb} />);
+
+      expect(getByRole('checkbox')).toHaveProperty('checked');
+      expect(getByRole('checkbox')).toHaveProperty('id', uniqueId);
+    });
   });
 
-  test('Checkbox without a defined id should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(<Checkbox label={'Label'} onChange={jest.fn()} />);
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered with extra, arbitrary props', () => {
+    it('should spread extra props', () => {
```

Comment:
Thinking about when it fails, it may help to indicate where it spreads to
```suggestion
    it('should spread extra props onto the checkbox', () => {
```